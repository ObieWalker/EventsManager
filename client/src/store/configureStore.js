import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;