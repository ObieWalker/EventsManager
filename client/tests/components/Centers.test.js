import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedCenter, { AllCenters } from '../../src/components/Centers.jsx';
import { centers } from '../__mocks__/centersData';

let mountedComponent;
const allCenters = {};
let props;

const getAllCenters = jest.fn(() => Promise.resolve({}));
// const getMoreCenters = jest.fn();
// const location = {};

const mockStore = configureStore();
let wrapper;

const getComponent = () => {
  if (!mountedComponent) {
    props = {
      allCenters,
      getAllCenters
    };
    mountedComponent = shallow(<AllCenters {...props} />);
  }
  return mountedComponent;
};

describe('Center Component', () => {
  beforeEach(() => {});

  it('component successfully rendered', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      allCenters: {
        fetchedCenters: [centers]
      }
    };

    wrapper = getComponent();
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().centers).toEqual(nextProps.allCenters.fetchedCenters);
    expect(wrapper.state().isLoading).toEqual(false);
  });

  it('testing load more function', () => {
    wrapper = getComponent();
    wrapper.instance().loadMoreContent();
    expect(wrapper.state().pageNo).toEqual(2);
    expect(wrapper.state().limit).toEqual(6);
  });


  describe('Connected Component', () => {
    describe('Connected Centers Component', () => {
      it('component successfully rendered', () => {
        const store = mockStore({
          centers
        });
        wrapper = shallow(<ConnectedCenter store={store} allCenters={allCenters} />);
        expect(wrapper.length).toBe(1);
      });
    });
  });
});
