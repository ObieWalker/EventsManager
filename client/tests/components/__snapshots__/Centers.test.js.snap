// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllCenters
    allCenters={Object {}}
    getAllCenters={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            6,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h3>
          All centers and details.
        </h3>
        <div
          className="center"
        >
          <Connect(Search) />
           
          <br />
           
          <br />
        </div>
        <div>
           
          There are no registered centers.
          <button
            className="btn btn-primary active"
            id="loadMore"
            onClick={[Function]}
          >
            Load More
          </button>
          <br />
          <br />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            All centers and details.
          </h3>,
          <div
            className="center"
          >
            <Connect(Search) />
             
            <br />
             
            <br />
          </div>,
          <div>
             
            There are no registered centers.
            <button
              className="btn btn-primary active"
              id="loadMore"
              onClick={[Function]}
            >
              Load More
            </button>
            <br />
            <br />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All centers and details.",
          },
          "ref": null,
          "rendered": "All centers and details.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Search) />,
              " ",
              <br />,
              " ",
              <br />,
            ],
            "className": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "There are no registered centers.",
              false,
              <button
                className="btn btn-primary active"
                id="loadMore"
                onClick={[Function]}
              >
                Load More
              </button>,
              <br />,
              <br />,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "There are no registered centers.",
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Load More",
                "className": "btn btn-primary active",
                "id": "loadMore",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Load More",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h3>
            All centers and details.
          </h3>
          <div
            className="center"
          >
            <Connect(Search) />
             
            <br />
             
            <br />
          </div>
          <div>
             
            There are no registered centers.
            <button
              className="btn btn-primary active"
              id="loadMore"
              onClick={[Function]}
            >
              Load More
            </button>
            <br />
            <br />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              All centers and details.
            </h3>,
            <div
              className="center"
            >
              <Connect(Search) />
               
              <br />
               
              <br />
            </div>,
            <div>
               
              There are no registered centers.
              <button
                className="btn btn-primary active"
                id="loadMore"
                onClick={[Function]}
              >
                Load More
              </button>
              <br />
              <br />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All centers and details.",
            },
            "ref": null,
            "rendered": "All centers and details.",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Search) />,
                " ",
                <br />,
                " ",
                <br />,
              ],
              "className": "center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "There are no registered centers.",
                false,
                <button
                  className="btn btn-primary active"
                  id="loadMore"
                  onClick={[Function]}
                >
                  Load More
                </button>,
                <br />,
                <br />,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "There are no registered centers.",
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Load More",
                  "className": "btn btn-primary active",
                  "id": "loadMore",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Load More",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
